cmd/control_loop/controller/controller.go

```diff
@@ -0,0 +1,200 @@
+package  controller
+
+import (
+    "github.com/apigee/registry/cmd/control_loop/list"
+    resource_lib "github.com/apigee/registry/cmd/control_loop/resources"
+    "reflect"
+    "github.com/apigee/registry/cmd/control_loop/manifest_reader"
+    "context"
+	"github.com/apigee/registry/connection"
+	"time"
+	"fmt"
+)
+
+type ResourceCollection struct {
+	maxUpdateTime time.Time
+	resources *[]resource_lib.Resource
+}
+
+func ProcessManifest(manifest *manifest_reader.Manifest) ([]string, error) {
+
+	var actions []string
+	for _, entry := range manifest.Entries {
+    	ctx := context.TODO()
+		client, err := connection.NewClient(ctx)
+		if err != nil {
+			return nil, err
+		}		
+		resourcePattern := fmt.Sprintf("projects/%s/%s", manifest.Project, entry.Resource)
+
+		// Process Dependencies
+		dependencyMaps := make([]map[string]*ResourceCollection, 0, 0)
+		err = processDependencies(
+			resourcePattern,
+			entry.Dependencies,
+			&dependencyMaps,
+			ctx,
+			client)
+		if err != nil {
+			return nil, err
+		}
+
+		cmds, err := evaluateResources(
+			resourcePattern,
+			entry.Dependencies,
+			dependencyMaps,
+			entry.Action,
+			ctx,
+			client)
+		if err != nil {
+			return nil, err
+		}
+		if len(cmds) > 0 {
+			actions = append(actions, cmds...)
+		}
+	}
+
+	return actions, nil
+}
+
+func processDependencies(
```

Comment:
Subjectively, I don't feel that "process" is the best word to use here, because it suggests to me that we're making some changes to the dependencies. If we're just checking the timestamps of our dependencies or building in-memory data that we will use to check timestamps, "checkDependencies" might be more clear to readers.

Reply:
Yep, processDependencies was a little unclear. Updated to `populateDependencyMaps`